prExplorer/Application/Products/Details.cs

```diff
@@ -26,8 +26,8 @@ public Handler(DataContext context, IMapper mapper)
 
             public async Task<Result<ProductDto>> Handle(Query request, CancellationToken cancellationToken)
             {
-                var productReturned = _mapper.Map<ProductDto>(await _context.Products.Where(x => x.Id == request.Id).FirstOrDefaultAsync());
-                return Result<ProductDto>.Success(productReturned);
+                var productsReturned = _mapper.Map<ProductDto>(await _context.Products.Where(x => x.Id == request.Id).FirstOrDefaultAsync());
+                return Result<ProductDto>.Success(productsReturned);
```

Comment:
This will return one product so productReturned instead of productsReturned will be better to use here.

Reply:
Checked and edited.